lookup_plugin/s3_handler.py

```diff
@@ -1071,6 +1063,26 @@ def perform_s3_operation(cluster_params, dirName, operation, chunk):
         bin_path = '%s/s3Operation' % binary_dir
         process = subprocess.Popen([bin_path, '-bucketName', 'paroscale-test', '-operation', operation, '-s3config', s3config, '-filepath', file_path, '-l', logFile])
 
+def Perform_S3_Operation(cluster_params, filename, operation, vdev, chunk, list_prefix):
+    base_dir = cluster_params['base_dir']
+    raft_uuid = cluster_params['raft_uuid']
+    binary_dir = os.getenv('NIOVA_BIN_PATH')
+    bin_path = '%s/s3Operation' % binary_dir
+    s3config = '%s/s3.config.example' % binary_dir
+    logFile = "%s/%s/s3operation" % (base_dir, raft_uuid)
+    cmd = [(bin_path, '-bucketName', 'paroscale-test', '-operation', operation, '-v', vdev, '-c', chunk, '-s3config', s3config, '-l', logFile)]
+
+    match operation:
+        case "list":
+            cmd.extend[( "-p", prefix)]
+        case "upload":
+            cmd.extend[( "-filepath", filename)]
+        case "delete":
+            cmd.extend[( "-f", filename, "-filepath", filename)]
```

Comment:
why do we have -p, -filepath, -f 
all are just for passing paths right.
can we just use 1 param -p to pass prefix incase of list and filename incase of upload and delete operations

Reply:
-p is prefix right, shall i combine that?  i am thinking we will only have -p for listing and -filepath for file

Reply:
both are for path only right, 
so a single parameter works for both

Reply:
Added  Changes as suggested!!